### PR TITLE
[TA] cleanup on detect language tests

### PR DESCRIPTION
**Pull request by maririos**
Changes in https://github.com/Azure/azure-sdk-for-net/issues/14833 for Detect Language

### CODE REVIEW

File: sdk/textanalytics/Azure.AI.TextAnalytics/tests/DetectLanguageTests.cs [hunk 1]
```diff
@@ -249,5 +232,50 @@ public async Task DetectLanguageBatchWithNullTextTest()
             InvalidOperationException ex = Assert.Throws<InvalidOperationException>(() => results[0].PrimaryLanguage.GetType());
             Assert.AreEqual(exceptionMessage, ex.Message);
         }
+
+        private void ValidateInDocumenResult(DetectedLanguage language)
+        {
+            Assert.That(language.Name, Is.Not.Null.Or.Not.Empty);
```
Comment (kinelski) [inline sdk/textanalytics/Azure.AI.TextAnalytics/tests/DetectLanguageTests.cs:238]: Should it be `Is.Not.Null.And.Not.Empty` instead? With `Or`, both null and empty will pass.

File: sdk/textanalytics/Azure.AI.TextAnalytics/tests/DetectLanguageTests.cs [hunk 2]
```diff
@@ -249,5 +232,50 @@ public async Task DetectLanguageBatchWithNullTextTest()
...
+            Assert.That(language.Iso6391Name, Is.Not.Null.Or.Not.Empty);
```
Comment (kinelski) [inline sdk/textanalytics/Azure.AI.TextAnalytics/tests/DetectLanguageTests.cs:239]: Same here about `Or`.

File: sdk/textanalytics/Azure.AI.TextAnalytics/tests/DetectLanguageTests.cs [hunk 3]
```diff
@@ -249,5 +232,50 @@ public async Task DetectLanguageBatchWithNullTextTest()
...
+            Assert.GreaterOrEqual(language.ConfidenceScore, 0.0);
+        }
+
+        private void ValidateBatchDocumentsResult(DetectLanguageResultCollection results, bool includeStatistics = default)
+        {
+            Assert.IsNotEmpty(results.ModelVersion);
```
Comment (kinelski) [inline sdk/textanalytics/Azure.AI.TextAnalytics/tests/DetectLanguageTests.cs:245]: Should we null-check here as well?

File: sdk/textanalytics/Azure.AI.TextAnalytics/tests/DetectLanguageTests.cs [hunk 4]
```diff
@@ -249,5 +232,50 @@ public async Task DetectLanguageBatchWithNullTextTest()
...
+                Assert.IsNotNull(results.Statistics);
+                Assert.Greater(results.Statistics.DocumentCount, 0);
+                Assert.Greater(results.Statistics.TransactionCount, 0);
+                Assert.GreaterOrEqual(results.Statistics.InvalidDocumentCount, 0);
+                Assert.GreaterOrEqual(results.Statistics.ValidDocumentCount, 0);
+            }
+            else
+                Assert.IsNull(results.Statistics);
+
+            foreach (DetectLanguageResult languageInDocument in results)
```
Comment (kinelski) [inline sdk/textanalytics/Azure.AI.TextAnalytics/tests/DetectLanguageTests.cs:258]: Is it necessary to assert that at least one result has been returned?

File: sdk/textanalytics/Azure.AI.TextAnalytics/tests/DetectLanguageTests.cs [hunk 5]
```diff
@@ -249,5 +232,50 @@ public async Task DetectLanguageBatchWithNullTextTest()
...
+            {
+                Assert.IsNotEmpty(languageInDocument.Id);
```
Comment (kinelski) [inline sdk/textanalytics/Azure.AI.TextAnalytics/tests/DetectLanguageTests.cs:260]: Should we null-check here?

**Review by kinelski — APPROVED** [review]

File: sdk/textanalytics/Azure.AI.TextAnalytics/tests/DetectLanguageTests.cs (hunk 3 above)
Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/tests/DetectLanguageTests.cs:245]: It is required on the swagger so codegen should help us there.

File: sdk/textanalytics/Azure.AI.TextAnalytics/tests/DetectLanguageTests.cs (hunk 4 above)
Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/tests/DetectLanguageTests.cs:258]: yes. good point. will add

File: sdk/textanalytics/Azure.AI.TextAnalytics/tests/DetectLanguageTests.cs (hunk 5 above)
Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/tests/DetectLanguageTests.cs:260]: It is required on the swagger so codegen should help us there.

File: sdk/textanalytics/Azure.AI.TextAnalytics/tests/DetectLanguageTests.cs (hunk 1 above)
Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/tests/DetectLanguageTests.cs]: updated

**Pull request merged**